### PR TITLE
fixes #7

### LINKED ISSUE #7: Make project_member_id not required for upload_file
When calling the `upload_file` function you have to give the `project_member_id` as the API call requires this. I think it would be nice if this could be optional as it only makes sense if you've already called `exchange_oauth2_member` on your own (or if you already have stored the `project_member_id`).

If you're not providing the `project_member_id` it could just run `exchange_oauth2_member` itself and get the `project_member_id` that way.

### PR DESCRIPTION
**Pull request by DartingMelody**
## General Checkups
- [x] Have you checked that there aren't other open pull requests for the same issue/update/change?
- [ ] If your code includes new features and not just bug fixes/copy editing: Did you include new tests?

### CODE REVIEW

**Comment by DartingMelody** [comment]
Hi @gedankenstuecke , please review the pr

**Comment by madprime** [comment]
Hi @DartingMelody, can you please merge `master` to bring this PR up-to-date with it, and fix all the style issues raised by Hound? Thanks!

**Comment by DartingMelody** [comment]
@madprime I have merged the master branch. I have incorporated the hound suggested changes but I can't see the violations explicitly. Help, please. Thanks.

**Comment by DartingMelody** [comment]
Hi, @madprime. I have resolved most hound issues. However, I am not able to see which single violation still remains. Please advise. Thanks.

File: ohapi/api.py [hunk 1]
```diff
@@ -138,9 +138,12 @@ def upload_file(target_filepath, metadata, access_token, project_member_id,
 
     logging.info('Uploading {} ({})'.format(
         target_filepath, format_size(filesize)))
+
```
[the remark this answers is not in the document]
Comment (gedankenstuecke) [reply ohapi/api.py]: Could you remove the spaces that are in this otherwise empty line? Once that's done hound should be all satisfied. 👍

**Comment by DartingMelody** [comment]
@gedankenstuecke  thanks for the help. Please review :)

**Comment by madprime** [comment]
Looks good, thanks!

**Pull request merged**